pkg/flow/internal/controller/custom_component_registry.go

```diff
@@ -28,6 +28,20 @@ func NewCustomComponentRegistry(parent *CustomComponentRegistry) *CustomComponen
 	}
 }
 
+func (s *CustomComponentRegistry) getDeclare(name string) (ast.Body, bool) {
+	s.mut.Lock()
+	defer s.mut.Unlock()
+	declare, ok := s.declares[name]
+	return declare, ok
+}
+
+func (s *CustomComponentRegistry) getImport(name string) (*CustomComponentRegistry, bool) {
+	s.mut.Lock()
+	defer s.mut.Unlock()
```

Comment:
nit, non-blocking: since this is only reading, could it be an RLock and RUnlock?
```suggestion
	s.mut.RLock()
	defer s.mut.RUnlock()
```